### PR TITLE
NT-0019 - SignIn route basic structure

### PR DESCRIPTION
**Pull request by kamilmoskal**

### CODE REVIEW

File: client/src/components/Button/Button.stories.tsx [hunk 1]
```diff
@@ -3,31 +3,31 @@ import React from 'react';
...
-    <Button size="large" onClick={action('clicked')}>
-      Hello Button
-    </Button>
-  ))
-  .add('primary', () => (
-    <Button primary onClick={action('clicked')}>
-      Hello Button
-    </Button>
-  ));
+// storiesOf('Button', module)
```
Comment (damian0o) [inline client/src/components/Button/Button.stories.tsx]: i believe we can remove this commented code

File: client/src/components/Button/Button.tsx [hunk 2]
```diff
@@ -1,72 +1,25 @@
...
-    scale: 1
-  },
-  hover: {
-    scale: 1.1
-  },
-  press: {
-    scale: 0.8
-  }
-});
+export const ButtonMUI:React.FC<ButtonProps & ButtonMUIProp> = ({ children, type, color, variant }) => {
```
Comment (damian0o) [inline client/src/components/Button/Button.tsx]: mui part is no longer required

File: client/src/containers/SignIn/SignInForm/SignInForm.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
+import React from "react";
+import FormHelperText from "@material-ui/core/FormHelperText";
+import styled from "styled-components";
+import { Formik, Form, Field } from "formik";
+import FormTextField from "../../../components/FormTextField/FormTextField";
```
Comment (damian0o) [inline client/src/containers/SignIn/SignInForm/SignInForm.tsx]: we should provide some better folder structure. 3 levels up i believe is to much. for now it is ok though.

File: client/src/containers/SignIn/SignInForm/SignInForm.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+interface Props {
+  onSubmit: (values: Values) => void;
+}
+
+const SignIn: React.FC<Props> = ({onSubmit}) => {
+  return (
+ 
+      <Formik
```
Comment (damian0o) [inline client/src/containers/SignIn/SignInForm/SignInForm.tsx]: how this is working with server side errors ?

File: client/src/containers/SignIn/SignInForm/SignInForm.tsx (hunk 3 above)
Comment (mras2303) [reply client/src/containers/SignIn/SignInForm/SignInForm.tsx]: There is a way to use absolute path

`import FormTextField from "components/FormTextField/FormTextField";`

File: client/src/components/Button/Button.stories.tsx [hunk 5]
```diff
@@ -3,31 +3,31 @@ import React from 'react';
...
+//       Hello Button
+//     </Button>
+//   ))
+//   .add('primary', () => (
+//     <Button primary onClick={action('clicked')}>
+//       Hello Button
+//     </Button>
+//   ));
 
-storiesOf('ButtonA', module).add('default', () => (
```
Comment (mras2303) [inline client/src/components/Button/Button.stories.tsx]: What does mean ButtonA? I dont understand what should this component do

File: client/src/containers/SignIn/SignIn.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import Grid from "@material-ui/core/Grid";
+
+const SignIn = () => {
+  return (
+    <Grid
+    container
+    direction="column"
+    alignItems="center"
+    justify="center"
+    style={{ minHeight: "100vh" }}
```
Comment (mras2303) [inline client/src/containers/SignIn/SignIn.tsx]: Please, don't use inline styling. Better extend this component by styled-components

File: client/src/containers/SignIn/SignInForm/SignInForm.tsx (hunk 4 above)
Comment (kamilmoskal) [reply client/src/containers/SignIn/SignInForm/SignInForm.tsx]: it hasn't build in feature for handle outside errors, but there are some workarounds which i'll try use

File: client/src/containers/SignIn/SignInForm/SignInForm.tsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        }}
+        validationSchema={SigninSchema}
+      >
+        {({ errors }) => (
+          <StyledFormikForm>
+            <Field name="email" error={errors.email} label="Email *" component={FormTextField}/>
+            <Field name="password" error={errors.password} label="Password *" component={FormTextField}/>
+            <Field type="checkbox" name="remember" label="Remember me" component={FormCheckboxField}/>
+            {Object.keys(errors).length > 0 && <FormHelperText error>{Object.values(errors).map(error => <p>{error}</p>)}</FormHelperText>}
+            <ButtonFormPrimaryMUI>Submit</ButtonFormPrimaryMUI>
```
Comment (mras2303) [inline client/src/containers/SignIn/SignInForm/SignInForm.tsx]: This name completely has no sense for me

`<Button primary>Submit</Button>`
will be great

File: client/src/components/FormCheckboxField/FormCheckboxField.tsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+import { FieldProps } from 'formik';
+import Checkbox from '@material-ui/core/Checkbox';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import {FormControlLabelProps} from '@material-ui/core/FormControlLabel/FormControlLabel';
+
+
+const FormCheckboxField: React.FC<FieldProps & FormControlLabelProps> = ({ field, label }) => {
```
Comment (mras2303) [inline client/src/components/FormCheckboxField/FormCheckboxField.tsx]: If this is checkbox, we all know that it is relating to form. Maybe just `checkbox` or something simple like that will be good?

**Pull request merged**